### ISSUE TITLE
bean-check doesn't produce error listing when using data cached via picklecache

### ISSUE THREAD

**Issue by blais** [beancount/beancount#92]
**[Original report](https://bitbucket.org/blais/beancount/issue/92) by Red Street (GitHub: [redstreet](https://github.com/redstreet)).**

----------------------------------------

Steps to reproduce:
1) Change: [this line](https://bitbucket.org/blais/beancount/src/42ecea188d8daf595c4338ac84b29eea003d5587/src/python/beancount/loader.py?at=default&fileviewer=file-view-default#loader.py-135) to:

```
if t2 - t1 > time_threshold or True:
```

to force creating the picklecache to reproduce the bug on a small example

2) Run attached file two or more times in succession:

```
#!shell
$ bean-check bean-check-cache-bug.bc
$ bean-check bean-check-cache-bug.bc
$ bean-check bean-check-cache-bug.bc

Expected output:
---------------------
The balancing error on the input file should show up each time bean-check is run.

Actual output:
---------------------
Instead, the error shows up the first time, but not thereafter. Removing the picklecache file fixes the problem:
$ bean-check bean-check-cache-bug.bc     
bean-check-cache-bug.bc:5:       Transaction does not balance: (51.90705 USD)

   2001-01-01 * "Purchase"
     Assets:Investments1  24.721 AAPL {1.05 USD}
     Assets:Bank1          25.95 USD            


$ bean-check bean-check-cache-bug.bc
$ bean-check bean-check-cache-bug.bc
$ rm .bean-check-cache-bug.bc.picklecache
$ bean-check bean-check-cache-bug.bc     
bean-check-cache-bug.bc:5:       Transaction does not balance: (51.90705 USD)

   2001-01-01 * "Purchase"
     Assets:Investments1  24.721 AAPL {1.05 USD}
     Assets:Bank1          25.95 USD            

```

**Comment by blais** [comment]
**Original comment by Red Street (GitHub: [redstreet](https://github.com/redstreet)).**

----------------------------------------

[Reference thread](https://groups.google.com/d/msg/beancount/xGlhz1W3Hig/R27pryXrCQAJ)

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

I'm able to reproduce.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #92: a minor bug with the loader cache whereby errors weren't getting emitted on a cache hit.

**Comment by blais** [comment]
**Original comment by Red Street (GitHub: [redstreet](https://github.com/redstreet)).**

----------------------------------------

Tested, works, thanks!
